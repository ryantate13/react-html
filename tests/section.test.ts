import {section} from '../src';

describe('section', () => {
    it('renders with children', () => {
        expect(section({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(section({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(section({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(section()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(section({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(section({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(section({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(section({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(section({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(section({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(section({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(section({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(section({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(section({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(section({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(section({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(section({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(section({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(section({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(section({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(section({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(section({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(section({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(section({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(section({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(section({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(section({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(section({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(section({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(section({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(section({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(section({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(section({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(section({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(section({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(section({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(section({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(section({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(section({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(section({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(section({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(section({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(section({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(section({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(section({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(section({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(section({translate: "no"})).toMatchSnapshot();
    });
});