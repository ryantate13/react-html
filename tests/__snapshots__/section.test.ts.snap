// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section renders with accessKey="a,b,c,d" 1`] = `
<section
  accessKey="a,b,c,d"
/>
`;

exports[`section renders with autoCapitalize="characters" 1`] = `
<section
  autoCapitalize="characters"
/>
`;

exports[`section renders with autoCapitalize="none" 1`] = `
<section
  autoCapitalize="none"
/>
`;

exports[`section renders with autoCapitalize="off" 1`] = `
<section
  autoCapitalize="off"
/>
`;

exports[`section renders with autoCapitalize="on" 1`] = `
<section
  autoCapitalize="on"
/>
`;

exports[`section renders with autoCapitalize="sentences" 1`] = `
<section
  autoCapitalize="sentences"
/>
`;

exports[`section renders with autoCapitalize="words" 1`] = `
<section
  autoCapitalize="words"
/>
`;

exports[`section renders with children 1`] = `
<section>
  TEST
</section>
`;

exports[`section renders with className="foo bar baz" 1`] = `
<section
  className="foo bar baz"
/>
`;

exports[`section renders with contentEditable="false" 1`] = `
<section
  contentEditable="false"
/>
`;

exports[`section renders with contentEditable="true" 1`] = `
<section
  contentEditable="true"
/>
`;

exports[`section renders with dir="auto" 1`] = `
<section
  dir="auto"
/>
`;

exports[`section renders with dir="ltr" 1`] = `
<section
  dir="ltr"
/>
`;

exports[`section renders with dir="rtl" 1`] = `
<section
  dir="rtl"
/>
`;

exports[`section renders with draggable="false" 1`] = `
<section
  draggable="false"
/>
`;

exports[`section renders with draggable="true" 1`] = `
<section
  draggable="true"
/>
`;

exports[`section renders with empty attributes 1`] = `<section />`;

exports[`section renders with hidden="false" 1`] = `
<section
  hidden={false}
/>
`;

exports[`section renders with hidden="true" 1`] = `
<section
  hidden={true}
/>
`;

exports[`section renders with id="TEST" 1`] = `
<section
  id="TEST"
/>
`;

exports[`section renders with inputMode="decimal" 1`] = `
<section
  inputMode="decimal"
/>
`;

exports[`section renders with inputMode="email" 1`] = `
<section
  inputMode="email"
/>
`;

exports[`section renders with inputMode="none" 1`] = `
<section
  inputMode="none"
/>
`;

exports[`section renders with inputMode="numeric" 1`] = `
<section
  inputMode="numeric"
/>
`;

exports[`section renders with inputMode="search" 1`] = `
<section
  inputMode="search"
/>
`;

exports[`section renders with inputMode="tel" 1`] = `
<section
  inputMode="tel"
/>
`;

exports[`section renders with inputMode="text" 1`] = `
<section
  inputMode="text"
/>
`;

exports[`section renders with inputMode="url" 1`] = `
<section
  inputMode="url"
/>
`;

exports[`section renders with is="foo" 1`] = `
<section
  is="foo"
/>
`;

exports[`section renders with itemID="https://google.com/" 1`] = `
<section
  itemID="https://google.com/"
/>
`;

exports[`section renders with itemProp="https://google.com/" 1`] = `
<section
  itemProp="https://google.com/"
/>
`;

exports[`section renders with itemRef="foo bar" 1`] = `
<section
  itemRef="foo bar"
/>
`;

exports[`section renders with itemScope="false" 1`] = `
<section
  itemScope={false}
/>
`;

exports[`section renders with itemScope="true" 1`] = `
<section
  itemScope={true}
/>
`;

exports[`section renders with itemType="https://google.com/" 1`] = `
<section
  itemType="https://google.com/"
/>
`;

exports[`section renders with lang="" 1`] = `
<section
  lang=""
/>
`;

exports[`section renders with lang="es-419" 1`] = `
<section
  lang="es-419"
/>
`;

exports[`section renders with slot="TEST" 1`] = `
<section
  slot="TEST"
/>
`;

exports[`section renders with spellCheck="false" 1`] = `
<section
  spellCheck="false"
/>
`;

exports[`section renders with spellCheck="true" 1`] = `
<section
  spellCheck="true"
/>
`;

exports[`section renders with style="[object Object]" 1`] = `
<section
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`section renders with tabIndex="1" 1`] = `
<section
  tabIndex={1}
/>
`;

exports[`section renders with title="TEST" 1`] = `
<section
  title="TEST"
/>
`;

exports[`section renders with translate="no" 1`] = `
<section
  translate="no"
/>
`;

exports[`section renders with translate="yes" 1`] = `
<section
  translate="yes"
/>
`;

exports[`section renders without attributes 1`] = `<section />`;
